generic_server.cc

```diff
@@ -363,7 +380,13 @@ future<> server::do_accepts(int which, bool keepalive, socket_address server_add
             fd.set_nodelay(true);
             fd.set_keepalive(keepalive);
             auto conn = make_connection(server_addr, std::move(fd), std::move(addr),
-                    _conns_cpu_concurrency_semaphore, {});
+                    _conns_cpu_concurrency_semaphore, std::move(units));
+            if (shed) {
+                _logger.debug("too many in-flight connection attempts: {}, connection dropped",
+                        _conns_cpu_concurrency_semaphore.waiters());
```

Comment:
I think this merits a warn level message (we're impacting the user directly) but with rate limiting.